StereoKitC/stereokit_ui.cpp

```diff
@@ -1592,7 +1592,8 @@ bool32_t ui_hslider_at_g(const C *id_text, float &value, float min, float max, f
 	}
 
 	if (button_state & button_state_active) {
-		float new_val = min + fminf(1, fmaxf(0, ((window_relative_pos.x-button_size.x/2)-finger_x) / (size.x-button_size.x)))*(max-min);
+		float pos_in_slider = fmin(1, fmax(0, ((window_relative_pos.x-button_size.x/2)-finger_x) / (size.x-button_size.x)));
+		N new_val = (N)min + (N)pos_in_slider*(N)(max-min);
 		if (step != 0) {
 			new_val = min + ((int)(((new_val - min) / step) + 0.5f)) * step;
```

Comment:
This `0.5f` is technically just for rounding, but to reduce unit conversion back and forth in this case, it should be better to say `(N)0.5` or something.

Reply:
Oh indeed!